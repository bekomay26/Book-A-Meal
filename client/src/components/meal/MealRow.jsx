import React from 'react';
import PropTypes from 'prop-types';
import CSSModules from 'react-css-modules';
import '../../assets/styles/mm.css';

const MealRow = ({ meal, selectMealEditBtn, selectMealDelBtn, mealGoesWith, mealOnTop, addExtra, extraOptId, selected }) => {
  const goes = mealGoesWith(meal.extras);
  const onTop = mealOnTop(meal.extras);
  let extraOptIdn = 0;
  const xtraItem = (extraTitle, type) => {
    extraOptIdn += 1;
    return (
      <div key={extraOptIdn - 1} data-key={extraOptIdn - 1}>
        <div className="col-10">{extraTitle}</div>
        <div className="col-2"><i onClick={(event) => selected(event, type)} className="fas fa-minus extra-minus" role='button' /></div>
      </div>
    );
  };
  // console.log(goes);
  const goesExt = goes.map(ext =>
    xtraItem(ext.title, 'goes'));
  const topExt = onTop.map(ext =>
    xtraItem(ext.title, 'OnTop'));
  // addExtra(y, 'goes');

  return (
    <tr>
      <td>{meal.id}</td>
      <td>{meal.title}</td>
      <td>category</td>
      <td>&#x20A6;{meal.price}</td>
      <td onClick={() => selectMealEditBtn(meal,goesExt,topExt)} className="edit"><i className="far fa-edit" /></td>
      <td onClick={() => selectMealDelBtn(meal)} ><i className="far fa-trash-alt remove"></i></td>
    </tr>)
};

MealRow.propTypes = {
  meal: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  })
    .isRequired,
  selectMealEditBtn: PropTypes.func.isRequired,
};

// const MealRowWithCSS = CSSModules(MealRow, styles, { allowMultiple: true });
// export default (MealRowWithCSS);
export default MealRow;
