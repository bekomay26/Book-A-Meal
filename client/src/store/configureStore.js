import { createStore, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';


const configureStore = initialState =>
  createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunk, reduxImmutableStateInvariant()),
  );

export default configureStore;
