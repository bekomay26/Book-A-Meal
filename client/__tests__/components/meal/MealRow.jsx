import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import MealRow from '../../../src/components/meal/MealRow';
import { getMealsResponse } from '../../__mocks__/mockMeal';
import { getExtrasResponse } from '../../__mocks__/mockExtras';

// import jsdom from 'jsdom';
// const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
// global.document = doc
// global.window = doc.defaultView
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({
  authReducer: {
    isAuthenticated: false,
    isCaterer: false,
    errors: {},
  },
});

const mealArray = getMealsResponse.meals;
const extraArray = getExtrasResponse.extras;

function setup() {
  const props = {
    selectMealEditBtn: jest.fn(),
    selectMealDelBtn: jest.fn(),
    mealGoesWith: jest.fn().mockReturnValue([]),
    mealOnTop: jest.fn().mockReturnValue([]),
    meal: mealArray[0],
    addExtra: jest.fn(),
    selectChange: jest.fn(),
    selected: jest.fn(),
    extraOptId: 0,
  };
  return shallow(<MealRow {...props} />, reactrouter.get());
}

function mountSetup() {
  const props = {
    selectMealEditBtn: jest.fn(),
    selectMealDelBtn: jest.fn(),
    mealGoesWith: jest.fn().mockReturnValue([]),
    mealOnTop: jest.fn().mockReturnValue([]),
    meal: mealArray[1],
    addExtra: jest.fn(),
    selectChange: jest.fn(),
    selected: jest.fn(),
    extraOptId: 0,
  };
  return mount(<Provider store={store}><MealRow {...props} /></Provider>, reactrouter.get());
}

describe('The Meal Row Component', () => {
  // it('should render meal row component correctly', () => {
  //   const wrapper = setup();
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });

  // it('doesnt calls handleChecked Function', () => {
  //   const wrapper = mountSetup();
  //   wrapper.setProps({
  //     mealGoesWith: jest.fn().mockReturnValue(extraArray),
  //   });
  //   wrapper.debug();
  //   const btn = wrapper.find('.extra-minus').at(0);
  //   btn.simulate('click');
  //   expect(wrapper.prop('selected')).toHaveBeenCalled();
  // });

  it('should open drawer when a menu meal is clicked', () => {
    const wrapper = mountSetup();
    wrapper.find('.edit').simulate('click');
    expect(wrapper.find('.edit').length).toEqual(1);
    wrapper.unmount();
  });

  it('should open drawer when a menu meal is clicked', () => {
    const wrapper = mountSetup();
    wrapper.find('.fa-trash-alt').simulate('click');
    expect(wrapper.find('.fa-trash-alt').length).toEqual(1);
    wrapper.unmount();
  });

  it('When extras list is empty', () => {
    const props = {
      selectMealEditBtn: jest.fn(),
      selectMealDelBtn: jest.fn(),
      mealGoesWith: jest.fn().mockReturnValue([]),
      mealOnTop: jest.fn().mockReturnValue([]),
      meal: mealArray[4], // No extras meal
      addExtra: jest.fn(),
      selectChange: jest.fn(),
      selected: jest.fn(),
      extraOptId: 0,
    };
    const wrapper = mount(<Provider store={store}><MealRow {...props} /></Provider>, reactrouter.get());
    expect(wrapper.find('.fa-trash-alt').length).toEqual(1);
    expect(wrapper.find('.edit').length).toEqual(1);
    wrapper.unmount();
  });
});

