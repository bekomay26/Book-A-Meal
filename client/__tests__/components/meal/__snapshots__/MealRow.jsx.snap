// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Meal Row Component should render meal row component correctly 1`] = `
<tr>
  <td>
    54
  </td>
  <td>
    gfhgh
  </td>
  <td>
    category
  </td>
  <td>
    â‚¦
    980
  </td>
  <td
    className="edit"
    onClick={[Function]}
  >
    <i
      className="far fa-edit"
    />
  </td>
  <td
    onClick={[Function]}
  >
    <i
      className="far fa-trash-alt remove"
    />
  </td>
</tr>
`;
